### PR TITLE
Making history.state to be returned as copy, not reference in IE 8,9

### PR DESCRIPTION
**Pull request by izhyvaiev**
This closes https://github.com/devote/HTML5-History-API/issues/84

### CODE REVIEW

File: history.js [hunk 1]
```diff
@@ -240,7 +240,7 @@
          */
         "state": {
             get: function() {
-                return stateStorage[windowLocation.href] || null;
+                return typeof stateStorage[windowLocation.href] === 'object' ? JSON.parse(JSON.stringify(stateStorage[windowLocation.href])) : (stateStorage[windowLocation.href] || null);
```
Comment (devote) [inline history.js]: Hi!
Write a few lines to get a better read. I would also like to see changes to the file `history.ielte7.js`
Thanks!

Comment (izhyvaiev) [inline history.js]: Hi.
Adding same changes in `history.ielte7.js` can't be made in the same way as in `history.js` as IE7- doesn't support JSON object. So some cloning function needs to be written. Do you have some preferences in way of cloning objects or you don't care?

Comment (izhyvaiev) [inline history.js]: Oh I see you integrated JSON object to IE7- version. Than OK - I'll make some changes there and reformat it to clearer look

**Comment by izhyvaiev** [comment]
@devote done

**Comment by devote** [comment]
Very good! Thanks!

**Pull request merged**